### PR TITLE
[1LP][RFR] preconfigured openshift sprout providers

### PR DESCRIPTION
**Pull request by izapolsk**
This PR contains a bunch of sprout changes which are necessary for turning on preconfigured openshift appliances.
In addition, it restores mfalesni's sprout.sh script which is convenient to kick off sprout locally for testing/development purpose.
Besides that, there is .gitignore file for sprout which is convenient for development.

### CODE REVIEW

**Comment by izapolsk** [comment]
it isn't big. Major part of this PR is mfalesni's script which don't need to be reviewed.

File: cfme/utils/appliance/__init__.py [hunk 1]
```diff
@@ -2434,30 +2434,48 @@ def from_provider(cls, provider_key, vm_name, name=None, **kwargs):
...
+        app_kwargs = kwargs.copy()
 
-        if 'hostname' in kwargs:
-            hostname = kwargs.pop('hostname')
-        else:
+        if 'hostname' not in app_kwargs:
+            def is_ip_available():
+                try:
+                    ip = provider.mgmt.get_ip_address(vm_name)
+                    if ip is None:
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: **Optional**

```python
return ip or False
```

File: cfme/utils/appliance/__init__.py [hunk 2]
```diff
@@ -2434,30 +2434,48 @@ def from_provider(cls, provider_key, vm_name, name=None, **kwargs):
...
+                        return False
+                    else:
+                        return ip
+                except AttributeError:
```
Comment (mshriver) [inline cfme/utils/appliance/__init__.py]: **Optional**

Consider using `wait_for(handle_exception=True)`, though it isn't limited to only AttributeError exception skipping and may be too broad.

File: cfme/utils/appliance/__init__.py (hunk 1 above)
Comment (mshriver) [reply cfme/utils/appliance/__init__.py]: Went ahead and added a commit for this.

**Pull request merged**